### PR TITLE
implement font settings

### PR DESCRIPTION
**Pull request by bil-elmoussaoui**
screenshot from 2017-02-27 15-57-48

Here you go! 
I've got only a small problem:  It's impossible to disable font size in Gtk FontButton, which means that the output of Gtk.FontButton is always "FontName FontSize" and i don't really know how to handle that except splitting the text by space and removing the latest element from the Array and joining the text again?
screenshot from 2017-02-27 15-59-15

### CODE REVIEW

File: src/Widgets/SettingsDialog.vala [hunk 1]
```diff
@@ -18,6 +18,7 @@ public class FeedReader.SettingsDialog : Gtk.Dialog {
     private Gtk.ListBox m_serviceList;
     private Gtk.Stack m_stack;
     private InfoBar m_errorBar;
+    private Gtk.HeaderBar headerbar;
```
Comment (jangernert) [inline src/Widgets/SettingsDialog.vala]: member variable should always start with "m_"

File: src/Widgets/AddPopover.vala [hunk 2]
```diff
@@ -62,13 +62,14 @@ public class FeedReader.AddPopover : Gtk.Popover {
...
+		var opmlLabel = new Gtk.Label(_("OPML File:"));
 		opmlLabel.expand = true;
-		m_chooser = new Gtk.FileChooserButton(_("Select OPML-file"), Gtk.FileChooserAction.OPEN);
+		m_chooser = new Gtk.FileChooserButton(_("Select OPML File"), Gtk.FileChooserAction.OPEN);
 		var filter = new Gtk.FileFilter();
 		filter.add_mime_type("text/x-opml");
 		m_chooser.set_filter(filter);
-		m_chooser.expand = true;
+		m_chooser.expand = false;
+		m_chooser.get_style_context().add_class("flat");
```
Comment (jangernert) [inline src/Widgets/AddPopover.vala]: why exactly was this change made?
The text feels off-positioned now, and I'm not a big fan of making everything flat.

Comment (bil-elmoussaoui) [inline src/Widgets/AddPopover.vala]: It's just the button looks so big in smaller windows (i will revert that)

File: src/Widgets/SettingsDialog.vala (hunk 1 above)
Comment (bil-elmoussaoui) [inline src/Widgets/SettingsDialog.vala]: @jangernert I will get that done right now :)

**Comment by bil-elmoussaoui** [comment]
@jangernert Alright, so for you it's better to remove the old font size that we used to have? and just using Gtk.FontButton for everything? 
Seems to be a good idea for me. 

And what do you think about splitting the font size from the font familly ?

**Comment by jangernert** [comment]
Yes, like I said, I think it's better to use the FontButton for both type and size.

I don't really understand what you mean with the splitting. Are you talking about the displayed string on the button itself or the return value from the button after the font has been changed? In that case you can create a FontDescription object with the from_string constructor and query all needed information from that object. Just look at the description over here:
https://valadoc.org/gtk+-3.0/Gtk.FontButton.get_font_name.html

**Comment by bil-elmoussaoui** [comment]
@jangernert My bad didn't saw the description on the Valadocs! i will get that ready in few minutes :)

**Comment by jangernert** [comment]
@bil-elmoussaoui I think I finished the vala-part of the code. There is only a single font gsetting left that saves the whole font-string of the FontButton. The string is parsed inside the buildArticle() function.

**Comment by bil-elmoussaoui** [comment]
@jangernert Please check my comment on your latest commit ;-)

**Comment by jangernert** [comment]
Fixed.

**Comment by jangernert** [comment]
Only thing left is the fontsize.
The size property of the description is a bit weird right now
```
[ DEBUG ] URW Chancery L Medium Italic 12
[ DEBUG ] fontsize: 12288
```

**Comment by bil-elmoussaoui** [comment]
@jangernert Because the font size is not absolute, check the documentation, i'm trying to figure out how to convert it to pixels.

**Comment by bil-elmoussaoui** [comment]
@jangernert + I was thinking about removing the font family defined on the css file so the one chosen by the user can override the rest?

**Comment by jangernert** [comment]
Okay, but we should keep the monospace font for code :)
Regarding the fontsize: I know, it should be in points, right? I checked get_size_is_absolute() and it returned false.

**Comment by bil-elmoussaoui** [comment]
Yeah i kept the monospace for code :)I got the same result

**Comment by jangernert** [comment]
Did a quick workaround.

**Comment by bil-elmoussaoui** [comment]
I got a better solution :) I will push it just right now

**Comment by jangernert** [comment]
Don't you need the "pt" at the end to make it work?

**Comment by jangernert** [comment]
nvm

**Comment by bil-elmoussaoui** [comment]
Can you test it and see if everything works as excepted in your case? The log output shows the right size here :)

**Comment by jangernert** [comment]
Size seems a bit too small over here. I had it at 12pt. Now 12pt is tiny and I have to go up to 16pt to have roughly the same size.

**Comment by bil-elmoussaoui** [comment]
@jangernert Well, we can change the default font size; and it's up to the user to decide which default font settings they want?

**Comment by jangernert** [comment]
That's not the point. Shouldn't be 12pt the same size independently from the application. With the latest commit 12pt in libreoffice is now the same size as 16pt in FeedReader.
Are you sure nothing went wrong when you convert to px? Also why not convert to pt?

**Comment by jangernert** [comment]
> When setting font sizes, device units are always considered to be points (as in "12 point font"), rather than pixels.

**Comment by jangernert** [comment]
Now everything seems to be in order.

**Comment by bil-elmoussaoui** [comment]
Great work on this ;) I will open a new PR for the articles theming tomorrow

**Comment by jangernert** [comment]
Well, most of the work was done by you sir :D

**Pull request merged**